src/app/clusters/mode-base-server/mode-base-server.cpp

```diff
@@ -0,0 +1,445 @@
+            return StatusIB(Protocols::InteractionModel::Status::InvalidCommand).ToChipError();
+        }
+
+        switch (attributePath.mAttributeId)
+        {
+        case ModeBase::Attributes::StartUpMode::Id:
+            UpdateStartUpMode(newMode);
+            return CHIP_NO_ERROR;
+        case ModeBase::Attributes::OnMode::Id:
+            UpdateOnMode(newMode);
+            return CHIP_NO_ERROR;
+        }
+    }
+
+    return StatusIB(Protocols::InteractionModel::Status::InvalidCommand).ToChipError();
+}
+
+void Instance::UpdateCurrentMode(uint8_t aNewMode)
+{
+    uint8_t oldMode = mCurrentMode;
+    mCurrentMode    = aNewMode;
+    if (mCurrentMode != oldMode)
+    {
+        // The Administrator Commissioning cluster is always on the root endpoint.
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::CurrentMode::Id);
+    }
+}
+
+void Instance::UpdateStartUpMode(DataModel::Nullable<uint8_t> aNewStartUpMode)
+{
+    DataModel::Nullable<uint8_t> oldStartUpMode = mStartUpMode;
+    mStartUpMode                                = aNewStartUpMode;
+    if (mStartUpMode != oldStartUpMode)
+    {
+        // The Administrator Commissioning cluster is always on the root endpoint.
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::StartUpMode::Id);
+    }
+}
+
+void Instance::UpdateOnMode(DataModel::Nullable<uint8_t> aNewOnMode)
+{
+    DataModel::Nullable<uint8_t> oldOnMode = mOnMode;
+    mOnMode                                = aNewOnMode;
+    if (mOnMode != oldOnMode)
+    {
+        // The Administrator Commissioning cluster is always on the root endpoint.
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::OnMode::Id);
+    }
+}
+
+DataModel::Nullable<uint8_t> Instance::GetStartUpMode() const
+{
+    return mStartUpMode;
+}
+
+DataModel::Nullable<uint8_t> Instance::GetOnMode() const
+{
+    return mOnMode;
+}
+
+uint8_t Instance::GetCurrentMode() const
+{
+    return mCurrentMode;
+}
+
+bool Instance::IsSupportedMode(uint8_t modeValue)
+{
+    for (uint8_t i = 0; i < NumberOfModes(); i++)
+    {
+        uint8_t value;
+        auto err = GetModeValueByIndex(i, value);
+        if (err == CHIP_NO_ERROR)
+        {
+            if (value == modeValue)
+            {
+                return true;
+            }
+        }
+        else
+        {
+            break;
+        }
+    }
+    ChipLogDetail(Zcl, "Cannot find a mode with value %u", modeValue);
+    return false;
+}
+
+uint8_t Instance::NumberOfModes()
+{
+    return 0;
+}
+
+CHIP_ERROR Instance::GetModeLabelByIndex(uint8_t modeIndex, MutableCharSpan & label)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR Instance::GetModeValueByIndex(uint8_t modeIndex, uint8_t & value)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR Instance::GetModeTagsByIndex(uint8_t modeIndex, List<ModeTagStructType> & modeTags)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
```

Comment:
These are all required for proper operation. Just make them pure virtual and remove the base class version.

Reply:
Done [here](https://github.com/project-chip/connectedhomeip/pull/27504/commits/be64a611f136960530ada54407c47964c0909176).